ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java

```diff
@@ -129,7 +129,16 @@ private PlanNode buildQueryLogicalPlan(
               ksqlStructuredDataOutputNode.getKsqlTopic()
           );
       if (analysis.isDoCreateInto()) {
-        tempMetaStore.putTopic(ksqlStructuredDataOutputNode.getKsqlTopic());
+        try {
+          tempMetaStore.putTopic(ksqlStructuredDataOutputNode.getKsqlTopic());
+        } catch (KsqlException e) {
+          final String sourceName = tempMetaStore.getSourceForTopic(
+              ksqlStructuredDataOutputNode.getKsqlTopic().getName()).getName();
+          throw new KsqlException(
```

Comment:
Is this the only reason why the `putTopic` can fail?

Reply:
I think the point is that we should make sure that the put failed because the topic already exists _before_ returning that in the error message. 

This way, if the put can fail in the future for other reasons, we won't return the wrong error message.

Reply:
The only time we get `KsqlException` from `putTopic` is when the topic already exists. For any other problem we won't get `KsqlException`. Also the check is done in the `putTopic` method anyway!